packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st

```diff
@@ -1,21 +1,18 @@
 events
 chatHistoryReceived: aChatHistory
 
-	| chat chatMessageNumber oldestMessageId chatID |
+	| chat chatID |
 
 	(aChatHistory at: 'messages') ifNotEmpty: [
-		chatID := ((aChatHistory at: 'messages') first at: 'chat_id').
+		chatID := (aChatHistory at: 'messages') first at: 'chat_id'.
 		chat := self app chats getChat: chatID.
 
-		(aChatHistory at: 'messages') do: [:message |
-			((message at: 'content') at: '@type') = 'messageText' ifTrue: [
-				chat addMessage: (TCCMessage new
-					message: (((message at: 'content') at: 'text') at: 'text');
-					chatID: chatID;
-					senderId: (message at: 'sender_user_id');
-					messageId: (message at: 'id'))]].
+		(aChatHistory at: 'messages') do: [:message | chat addMessage: (TCCMessage newFromMessageEvent: message)].
 
-		chat messages ifNotEmpty: [
-			oldestMessageId := chat messages last messageId.
-			chatMessageNumber := self messageLimit - chat messages size.
-			chatMessageNumber > 0 ifTrue: [self getChatHistoryFrom: chatID with: oldestMessageId and: chatMessageNumber]]].
\ No newline at end of file
+		chat waitingForUpdate: false.
+		
+		chat stillRequestedMessages > 0 
+			ifTrue: [self 
```

Comment:
spacing looks wrong

Reply:
self in new line?

Reply:
not sure bout that but the stuff after self should be tabbed in